packages/block-library/src/navigation-menu/edit.js

```diff
@@ -164,7 +225,13 @@ function NavigationMenu( {
 
 export default compose( [
 	withColors( { backgroundColor: 'background-color', textColor: 'color' } ),
-	withSelect( ( select ) => {
+	withSelect( ( select, ownProps, registry ) => {
+		const { clientId } = ownProps;
+		const { getBlocks } = registry.select( 'core/block-editor' );
```

Comment:
Why consume `registry` instead of accessing `select` directly?

```js
withSelect( ( select, { clientId } ) => {
  const innerBlocks = select( 'core/block-editor' ).getBlocks( clientId );
  …
```

Additional notes:

1. Calling `select( store ).selectorName( … )` is more idiomatic in Gutenberg than extracting `const selectorName` beforehand.
1. The object destructuring near `{ clientId }` is just personal preference.

Reply:
@aduth Once told me that `getBlocks` was expensive and access directly from registry could help mitigate performance concerns. Indeed, this is what we do in the Columns Block (which is where I think he told me about this).

Happy to avoid if that's no longer the case or I misunderstood.

Reply:
Simplified and fixed other items. You're right it was additional "noise" and complexity for no tangible benefit to the readability of the code.

Reply:
> aduth Once told me that `getBlocks` was expensive and access directly from registry could help mitigate performance concerns. Indeed, this is what we do in the Columns Block (which is where I think he told me about this).

Is this the comment in question? https://github.com/WordPress/gutenberg/pull/13899#discussion_r279876963

If so, and looking at the context, including the description of #11851, it sounds like this is about grabbing state to be used in actions via `withDispatch` and avoiding pulling that action-only state with another `withSelect`. Quote: "It is useful when you need to fetch some dynamic data from the store at the time when the event is fired, but at the same time, you never use it to render your component."

If I interpreted the above correctly, there is no benefit to consuming `registry`.

Reply:
Ah yes. You are right. I guess blindly follow advice isn't the best strategy even with the best of intentions. Thanks for flagging. Fixed.